src/master/metadata_backend_file.cc

```diff
@@ -238,16 +238,18 @@ uint8_t MetadataBackendFile::fs_storeall(DumpType dumpType) {
 
 static bool xattr_load(MetadataLoader::Options options) {
 	const uint8_t *ptr;
-	inode_t inode;
-	uint8_t attributeNameLength;
-	uint32_t attributeValueLength;
+	inode_t inode = 0;
+	uint8_t attributeNameLength = 0;
+	uint32_t attributeValueLength = 0;
 	XAttributeInodeEntry *xattrInodeEntry = nullptr;
 
 	while (true) {
+		xattrInodeEntry = nullptr;  // Reset pointer to avoid stale values
+
 		try {
 			ptr = options.metadataFile->seek(options.offset);
 		} catch (const std::exception &e) {
-			safs_pretty_syslog(LOG_ERR, "loading xattr: can't read xattr");
+			safs::log_err("loading xattr: can't read xattr");
```

Comment:
Consider using `safs::log_exception`

```suggestion
			safs::log_exception(e, "loading xattr: can't read xattr");
```

Reply:
Applied.